mantidimaging/core/rotation/polyfit_correlation.py

```diff
@@ -26,22 +25,40 @@ def do_calculate_correlation_err(store: np.ndarray, search_index: int, p0_and_18
 
 
 def find_center(images: ImageStack, progress: Progress) -> tuple[ScalarCoR, Degrees]:
-    # assume the ROI is the full image, i.e. the slices are ALL rows of the image
+    if images is None or images.proj180deg is None:
+        raise ValueError("images and images.proj180deg cannot be None")
+    # Assume the ROI is the full image, i.e. the slices are ALL rows of the image
     slices = np.arange(images.height)
-    shift = pu.create_array((images.height, ))
-
+    shift = pu.create_array((images.height, ), dtype=np.float32)
     search_range = get_search_range(images.width)
-    min_correlation_error = pu.create_array((len(search_range), images.height))
-    shared_search_range = pu.create_array((len(search_range), ), dtype=np.int32)
-    shared_search_range.array[:] = np.asarray(search_range, dtype=np.int32)
-    _calculate_correlation_error(images, shared_search_range, min_correlation_error, progress)
+    search_range_list = list(search_range)
+    min_correlation_error = pu.create_array((len(search_range), images.height), dtype=np.float32)
+    shared_search_range = pu.create_array((len(search_range_list), ), dtype=np.int32)
+    shared_search_range.array[:] = np.asarray(search_range_list, dtype=np.int32)
+    # Copy projections to shared memory
+    shared_projections = pu.create_array((2, images.height, images.width), dtype=np.float32)
+    shared_projections.array[0][:] = images.projection(0)
+    shared_projections.array[1][:] = np.fliplr(images.proj180deg.data[0])
 
+    # Convert SharedArray to SharedArrayProxy
+    shared_projections_proxy = shared_projections.array_proxy
+    shared_search_range_proxy = shared_search_range.array_proxy
+
+    # Run compute function parallel
+    params = {
+        'shared_projections': shared_projections_proxy,
+        'shared_search_range': shared_search_range_proxy,
+        'image_width': images.width,
+    }
+    ps.run_compute_func(compute_correlation_error,
+                        len(search_range_list), [min_correlation_error],
+                        params,
+                        progress=progress)
```

Comment:
Rather than dealing with the proxys manually, you can pass them into `run_compute_func` in the arrays argument:
```
    ps.run_compute_func(compute_correlation_error,
                        len(search_range_list),
                        [min_correlation_error, shared_projections, shared_search_range],
                        params,
                        progress=progress)
```
`params` is just for simple params line numbers or strings, that can safely be passed between processes. `arrays` handle all the difficult bits of sharing large arrays between processes.